mlflow/pyspark/ml/__init__.py

```diff
@@ -390,10 +395,17 @@ def _get_warning_msg_for_fit_call_with_a_list_of_params(estimator):
 
 def _get_tuning_param_maps(param_search_estimator, uid_to_indexed_name_map):
     tuning_param_maps = []
+
+    def gen_log_key(k):
+        if k.parent not in uid_to_indexed_name_map:
+            raise ValueError(
+                "tuning params should not include params not owned by the tuned estimator, but "
+                f"get a param which parent is {k.parent}"
```

Comment:
```suggestion
                "Tuning params should not include params not owned by the tuned estimator, but "
                f"found a param with parent {k.parent}"
```

Reply:
Can we include parameter names in the error message so that we can tell which parameters are invalid?

Reply:
Can we use the string reprsentation of a `Param` object?

https://github.com/apache/spark/blob/44b7931936d9eff4d8f3054abdef3363af26afb6/python/pyspark/ml/param/__init__.py#L55

```python
    def __str__(self):
        return str(self.parent) + "__" + self.name
```